app/ts/store/account.ts

```diff
@@ -1,87 +1,86 @@
-import { effect, Signal, useSignal } from '@preact/signals'
+import { Signal, useSignal } from '@preact/signals'
 import { ethers } from 'ethers'
-import { AsyncProperty, useAsyncState } from '../library/preact-utilities.js'
-import { assertsExternalProvider, isEthereumObservable } from '../library/utilities.js'
-
-export type ConnectMutation = {
-	transport: Signal<AsyncProperty<string>>
-	dispatch: () => void
-	reset: () => void
-}
-
-export type ReconnectMutation = {
-	transport: Signal<AsyncProperty<unknown>>
-	dispatch: () => void
-	reset: () => void
-}
+import { ConnectAttemptError } from '../library/exceptions.js'
+import { useAsyncState } from '../library/preact-utilities.js'
+import { assertsExternalProvider, assertUnreachable, isEthereumObservable } from '../library/utilities.js'
 
 type Account =
 	| {
-			isConnected: false
-			connectMutation: ConnectMutation
-			reconnectMutation: ReconnectMutation
+			state: 'disconnected'
+			connect: (attemptOnly?: boolean) => void
 	  }
 	| {
-			isConnected: true
+			state: 'connecting'
+			reset: () => void
+	  }
+	| {
+			state: 'connected'
 			address: string
 	  }
+	| {
+			state: 'failed'
+			error: Error
+	  }
 
 export type AccountStore = Signal<Account>
 export function createAccountStore() {
-	const { value: asyncValue, waitFor, reset } = useAsyncState<string>()
-	const reconnect = useAsyncState()
+	const { value: query, waitFor, reset } = useAsyncState<string>()
 
-	const reconnectMutation = {
-		transport: reconnect.value,
-		dispatch: () => {
-			reconnect.waitFor(async () => {
+	const connect = (attemptOnly?: boolean) =>
+		waitFor(async () => {
+			if (attemptOnly) {
 				try {
 					assertsExternalProvider(window.ethereum)
 					const provider = new ethers.providers.Web3Provider(window.ethereum)
 					const signer = provider.getSigner()
-					const address = await signer.getAddress()
-					accountStore.value = { isConnected: true, address }
-				} catch {
-					accountStore.value = accountDefaults
+					return await signer.getAddress()
+				} catch (error) {
+					throw new ConnectAttemptError()
```

Comment:
we could try to distinguish different errors here and if they don't match the ones we expect, we throw the error at it is

Reply:
The try-catch here is an attempt to verify if a wallet account is accessible and is expected to withdraw silently if the attempt failed to assume no account is connected yet. nevertheless, `ConnectAttemptError` will now receive the error message